### PR TITLE
acquisitions: update order and order lines resources

### PR DESCRIPTION
**Pull request by zannkukai**
* Updates `order` and `order_lines` schema and ES mappings.
* Adds order line information when indexing an order.
* Adds account remaining balance check when an order line is
  created or updated.
* Configures facets for orders.
* Closes rero/rero-ils#851.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/acq_orders/jsonschemas/acq_orders/acq_order-v0.0.1.json
Comment (benerken) [inline rero_ils/modules/acq_orders/jsonschemas/acq_orders/acq_order-v0.0.1.json:100]: Why do you replace the list of currencies by the free text of length 1 ???

File: rero_ils/modules/acq_orders/models.py
Comment (benerken) [inline rero_ils/modules/acq_orders/models.py:50]: PENDING has been removed no ? 
It's "approved" if I'm right.

File: tests/ui/acq_orders/test_acq_orders_api.py
Comment (benerken) [inline tests/ui/acq_orders/test_acq_orders_api.py:34]: Should  be  assert acor.status == AcqOrderStatus.APPROVED no ?

File: rero_ils/modules/acq_orders/api.py
Comment (zannkukai) [reply rero_ils/modules/acq_orders/api.py:112]: Nope. An order cannot have an "approved" status, it's non-sense. 
"Approved" tells that user execute an action on the resource to approve it. For an order it's not the case, we create an order_line (we can considerate this action as an approval) then the order is update but not really approved.

This is a vocabulary 'fight' without consequence on data and workflow.

File: rero_ils/modules/acq_orders/jsonschemas/acq_orders/acq_order-v0.0.1.json
Comment (zannkukai) [reply rero_ils/modules/acq_orders/jsonschemas/acq_orders/acq_order-v0.0.1.json:100]: because we could use any currency than vendor define. We need to use a shared currency table between vendor and orders.  If we maintain two list in parallel, it will be bad very soon :-)

File: rero_ils/modules/acq_orders/models.py
Comment (zannkukai) [reply rero_ils/modules/acq_orders/models.py:50]: APPROVED for order_line ; PENDING for order

File: rero_ils/modules/acq_accounts/models.py
Comment (iGormilhit) [inline rero_ils/modules/acq_accounts/models.py:46]:
```suggestion
    """Type of exceedance about an Acquisition account."""
```

File: rero_ils/modules/acq_order_lines/extensions.py
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/extensions.py:19]:
```suggestion
"""Extension for Acquisition account records."""
```

File: rero_ils/modules/acq_orders/models.py
Comment (iGormilhit) [inline rero_ils/modules/acq_orders/models.py:46]:
```suggestion
    """Available statuses about an Acquisition order."""
```

File: tests/ui/acq_order_lines/test_acq_order_lines_api.py
Comment (iGormilhit) [inline tests/ui/acq_order_lines/test_acq_order_lines_api.py:19]:
```suggestion
"""Acquisition order lines API tests."""
```

File: tests/ui/acq_orders/test_acq_orders_api.py
Comment (iGormilhit) [inline tests/ui/acq_orders/test_acq_orders_api.py:19]:
```suggestion
"""Acquisition orders API tests."""
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit:

- Component in the plural form.
- Missing trailing dot at the end of the last list item of the body.

**Review by iGormilhit — APPROVED** [review]

File: rero_ils/jsonschemas/common/currency-v0.0.1.json
Comment (rerowep) [inline rero_ils/jsonschemas/common/currency-v0.0.1.json:6]: Do we have to accept only this currencies?

Comment (zannkukai) [reply rero_ils/jsonschemas/common/currency-v0.0.1.json:6]: At this time, there are no PO demand to add other currency. By the way, it's now easy and centralized to add a currency

File: rero_ils/modules/acq_order_lines/api.py
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/api.py:115]:
```suggestion
        """Shortcut to the document object related to this order line."""
```

File: rero_ils/modules/acq_order_lines/extensions.py
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/extensions.py:68]: I don't get the context, but should the message refer to what it's trying to link to an harvested document?

File: rero_ils/modules/acq_order_lines/listener.py
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/listener.py:19]:
```suggestion
"""Signal connector for acquisition order line."""
```

File: rero_ils/modules/acq_order_lines/listener.py
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/listener.py:32]:
```suggestion
    :param doc_type: The document type of the record.
```
As long as it is documentation, code could be avoided. But maybe `doc_type` is clearer, your call. :slightly_smiling_face:

File: rero_ils/modules/acq_orders/listener.py
Comment (iGormilhit) [inline rero_ils/modules/acq_orders/listener.py:19]:
```suggestion
"""Signal connector for acquisition order."""
```

File: rero_ils/modules/acq_orders/listener.py
Comment (iGormilhit) [inline rero_ils/modules/acq_orders/listener.py:32]:
```suggestion
    :param doc_type: The document type of the record.
```

File: rero_ils/modules/acq_orders/models.py
Comment (iGormilhit) [inline rero_ils/modules/acq_orders/models.py:46]:
```suggestion
    """Available statuses for an acquisition order."""
```

File: rero_ils/modules/serializers.py
Comment (iGormilhit) [inline rero_ils/modules/serializers.py:106]:
```suggestion
        """Complete a bucket by adding a new key based on resource attributes.
```
Or
```suggestion
        """Complete a bucket by adding new keys based on resource attributes.
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message: `* Closes rero#851.` → `* Closes #851.`

File: rero_ils/modules/acq_order_lines/listener.py
Comment (zannkukai) [reply rero_ils/modules/acq_order_lines/listener.py:19]: There is only one signal at this time, but the file could contains several signals. So I keep the 's' ;-)

**Review by rerowep — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Pull request merged**
